src/plugins/controls/public/range_slider/components/range_slider_control.tsx

```diff
@@ -6,117 +6,57 @@
+  }, [debouncedOnChange, displayedValue]);
 
-  // EuiDualRange can only handle integers as min/max
-  const roundedMin = hasAvailableRange ? Math.floor(minValue) : minValue;
-  const roundedMax = hasAvailableRange ? Math.ceil(maxValue) : maxValue;
+  useEffect(() => {
+    setDisplayedValue(value ?? ['', '']);
+  }, [value]);
 
   const button = (
-    <button
-      onClick={() => setIsPopoverOpen(!isPopoverOpen)}
-      className="rangeSliderAnchor__button"
-      data-test-subj={`range-slider-control-${id}`}
-    >
-      <EuiFlexGroup gutterSize="none" responsive={false}>
-        <EuiFlexItem>
-          <EuiFieldNumber
-            controlOnly
-            fullWidth
-            className={`rangeSliderAnchor__fieldNumber ${
-              hasLowerBoundSelection && isInvalid ? INVALID_CLASS : ''
-            }`}
-            value={hasLowerBoundSelection ? lowerBoundValue : ''}
-            onChange={(event) => {
-              rangeSlider.dispatch.setSelectedRange([
-                event.target.value,
-                isNaN(upperBoundValue) ? '' : String(upperBoundValue),
-              ]);
-            }}
-            disabled={isLoading}
-            placeholder={`${hasAvailableRange ? roundedMin : ''}`}
-            isInvalid={isInvalid}
-            data-test-subj="rangeSlider__lowerBoundFieldNumber"
-          />
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EuiText className="rangeSliderAnchor__delimiter" size="s" color="subdued">
-            →
-          </EuiText>
-        </EuiFlexItem>
-        <EuiFlexItem>
-          <EuiFieldNumber
-            controlOnly
-            fullWidth
-            className={`rangeSliderAnchor__fieldNumber ${
-              hasUpperBoundSelection && isInvalid ? INVALID_CLASS : ''
-            }`}
-            value={hasUpperBoundSelection ? upperBoundValue : ''}
-            onChange={(event) => {
-              rangeSlider.dispatch.setSelectedRange([
-                isNaN(lowerBoundValue) ? '' : String(lowerBoundValue),
-                event.target.value,
-              ]);
-            }}
-            disabled={isLoading}
-            placeholder={`${hasAvailableRange ? roundedMax : ''}`}
-            isInvalid={isInvalid}
-            data-test-subj="rangeSlider__upperBoundFieldNumber"
-          />
-        </EuiFlexItem>
-        {isLoading ? (
-          <EuiFlexItem
-            grow={false}
-            className="rangeSliderAnchor__spinner"
-            data-test-subj="range-slider-loading-spinner"
-          >
-            <EuiLoadingSpinner />
-          </EuiFlexItem>
-        ) : null}
-      </EuiFlexGroup>
-    </button>
+    <RangeSliderButton
+      value={displayedValue}
+      onChange={setDisplayedValue}
+      onClick={(event) => {
+        // the popover should remain open if the click target is one of the number inputs
+        if (isPopoverOpen && event.target instanceof HTMLInputElement) {
+          return;
+        }
```

Comment:
This is meant to replicate the behaviour of the `EuiDualRange` component:

![Jun-12-2023 09-45-15](https://github.com/elastic/kibana/assets/8698078/eca04d80-5c20-4fb3-aa90-759859a69d5c)

We will be able to clean this up once we are officially able to make the transition :)